app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -127,4 +137,14 @@ class BrowserWebViewClient @Inject constructor(
                 }
                 .blockingFirst()
     }
+
+    private fun loadUrlOnUIThread(webView: WebView, url: String) {
+
+        Observable.just(url)
```

Comment:
Would be more readable to use the `post` method here to run on the UI thread i reckon

`webView.post({webView.loadUrl(url)})`

https://developer.android.com/reference/android/view/View.html#post(java.lang.Runnable)

Reply:
I think the `post` approach above is probably best here, but in general, if you needed something like this again and didn't have a view on which to `post`, you could use this type of Rx to avoid the awkward `just` and `map` usage:

     Completable.fromAction({ webView.loadUrl(url) })
                .subscribeOn(AndroidSchedulers.mainThread())
                .subscribe()

Reply:
Thanks @CDRussell - got my head so far up RX totally forgot about the standard stuff for handling the UI thread :)

Thanks for the completable tip, too.